Fix storage rtos types - remove including internal header file

0xc0170 opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html (Pull request template)
-->

Remove internal types from our storage rtos header file. And do some refactor (some files still use internal types). Plus some other fixes as I removed one internal rtx from our include, I discovered 2 files not including needed header files!

As result, this also fix some warnings about redefinition STRINGIFY for some platforms (nordic at least).

@JonatanAntoni who I discussed the issue with

### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [X] Fix
    [ ] Refactor
    [ ] New target
    [ ] Feature
    [ ] Breaking change

0xc0170 committed:
rtos: use RTX identifier

os_thread_t and family are internal and should not be used (thus we included
internal header file). Instead, use those that are exposed via rtx_os.h file.

0xc0170 committed:
device key: add memset include

0xc0170 committed:
lwip: fix rtx storage types

0xc0170 committed:
Thread: fix rtx thread type

0xc0170 committed:
kinetis: fix rtos storage type for thread

0xc0170 committed:
test: fix thread storage size

0xc0170 commented:
Fixed few errors (I noticed those internal types are used across the code base). This PR should clean it all

pingdan32 commented on rtos/TARGET_CORTEX/mbed_rtos_storage.h:
```diff
@@ -40,17 +40,17 @@ extern "C" {
  */
 
-#include "rtx_lib.h"
+#include "rtx_os.h"
```
Looks good. I like it.

marcuschangarm commented:
@TacoGrandeTX FYI

0xc0170 committed:
emac targets: fix thread cb type

0xc0170 commented:
One more fix incoming, another internal symbol not exposed (jenkins failure). This is a good exercise !

Update: more than one. More internal symbols in our codebase. Fixing it all in once! 🔥 🚒

0xc0170 committed:
boot: fix libspace num macro

RTX Config header file is internal and not exposed. With latest fixes to use
only public header files from RTX, we need to add these 2 new macros for
RTX ARMC lib configuration.

0xc0170 committed:
tests: fix internal symbols from RTX

As we do not include rtx_lib header file anymore, these symbols are not available.
Use core util for if ISR is active. And for OS tick, pull in os_tick header file.

0xc0170 committed:
test: fix os tick linkage

os_tick header file is C only, mangling would cause problems here

bulislaw commented on rtos/TARGET_CORTEX/mbed_boot.c:
```diff
@@ -430,6 +430,18 @@ void __rt_entry (void) {
  */
 #if defined(RTX_NO_MULTITHREAD_CLIB)
 
+// These 2 macros are taken from RTX Config (internal header)
```
That should probably go into our config file?

0xc0170 replied:
I can move it there with default values set. Based on our config, there would be just one
`#define OS_THREAD_LIBSPACE_NUM      4`, will check

0xc0170 replied:
Latest rebase fixed it, moved to conf with default value as it was (4)

marcuschangarm commented:
@0xc0170 
> Update: more than one. More internal symbols in our codebase. Fixing it all in once! 🔥 🚒
 
You are on a roll! 😄

deepikabhavnani reviewed (approved):
👍

cmonr commented:
/morph build

0xc0170 commented:
uvisor example failed - @alzix shall I update the example to fix the error or?

Error:

```
[DEBUG] Output: ./source/led3.cpp:65:5: error: 'os_thread_t' was not declared in this scope
[DEBUG] Output:      os_thread_t thread_def = {0};
```

0xc0170 commented:
Fix for uvisor example proposed https://github.com/ARMmbed/mbed-os-example-uvisor-thread/pull/104

Update: PR merged

0xc0170 commented:
/morph build

0xc0170 commented:
Waiting for final reviews for this bugfix. I labeled is as 5.10 due as apps could use these internal types we exposed. By 5.10, the types should be updated to the one we are providing as storage types for rtos.

cmonr commented:
+1 on targeting 5.10 due to the changes. LGTM.
